app/controllers/sessions_controller.rb

```diff
@@ -2,54 +2,84 @@ class SessionsController < ApplicationController
   def new
   end
 
+  def shop_owner_login
+  end
+
   def create
     if request.env["omniauth.auth"]
       oauth_user = OauthUser.new(request.env["omniauth.auth"])
       user = oauth_user.login_or_create
       session[:user_id] = user.id
 
       return redirect_to_user_intended unless session[:user_intended].nil?
-      redirect_to root_path, notice: "Welcome back, #{current_user.first_name}"
+      redirect_to root_path, notice: welcome_user
     else
-      process_form_login
+      process_form_login "RegularUser"
```

Comment:
@andela-tfowotade why are we differentiating them and how, cos I don't see a model called `RegularUser`